src/ahp/prover.rs

```diff
@@ -347,23 +314,29 @@ impl<F: PrimeField> AHPForR1CS<F> {
             })
             .collect();
 
+        let (f1, f2, f3) = rng
+            .squeeze_field_elements(3)
+            .iter()
+            .map(|x: &F| x.to_owned())
+            .collect_tuple()
+            .unwrap();
```

Comment:
This is not secure; we cannot use the sponge for randomness for zero knowledge. The old version should work fine.